addons/hr_expense/models/hr_expense.py

```diff
@@ -312,3 +304,161 @@ def action_get_attachment_view(self):
+            return 'hr_expense.mt_expense_approved'
+        elif 'state' in init_values and self.state == 'submit':
+            return 'hr_expense.mt_expense_confirmed'
+        elif 'state' in init_values and self.state == 'cancel':
+            return 'hr_expense.mt_expense_refused'
+        return super(HrExpenseSheet, self)._track_subtype(init_values)
+
+    def _add_followers(self):
+        user_ids = []
+        employee = self.employee_id
+        if employee.user_id:
+            user_ids.append(employee.user_id.id)
+        if employee.parent_id:
+            user_ids.append(employee.parent_id.user_id.id)
+        if employee.department_id and employee.department_id.manager_id and employee.parent_id != employee.department_id.manager_id:
+            user_ids.append(employee.department_id.manager_id.user_id.id)
+        self.message_subscribe_users(user_ids=user_ids)
+
+    @api.depends('expense_line_ids')
+    def _compute_amount(self):
+        for sheet in self:
+            total = 0.0
+            for line in sheet.expense_line_ids:
+                total += line.total_amount
+            sheet.total_amount = total
+
+    @api.multi
+    def _compute_payment_mode(self):
+        for sheet in self:
+            if sheet.expense_line_ids:
+                sheet.payment_mode = sheet.expense_line_ids[0].payment_mode
+
+    @api.multi
+    def _compute_attachment_number(self):
+        for sheet in self:
+            total = 0.0
+            for line in sheet.expense_line_ids:
+                total += line.attachment_number
+            sheet.attachment_number = total
+
+    @api.multi
+    def refuse_expenses(self, reason):
+        self.write({'state': 'cancel'})
+        if self.employee_id.user_id:
+            body = (_("Your Expense %s has been refused.<br/><ul class=o_timeline_tracking_value_list><li>Reason<span> : </span><span class=o_timeline_tracking_value>%s</span></li></ul>") % (self.name, reason))
+            self.message_post(body=body, partner_ids=[self.employee_id.user_id.partner_id.id])
+
+    @api.multi
+    def reset_expenses(self):
+        return self.write({'state': 'draft'})
+
+    @api.multi
+    def submit_expense_sheets(self):
+        self.write({'state': 'submit'})
+
+    @api.multi
+    def approve_expense_sheets(self):
+        self.message_post(body=_("The expense has been validated by %s") % (self.env['res.users'].browse(self._uid).name))
+        self.write({'state': 'approve', 'user_valid': self._uid})
+
+    @api.multi
+    def refuse_expense_sheets(self, reason):
+        self.write({'state': 'cancel'})
+        if self.employee_id.user_id:
+            body = (_("Your Expense %s has been refused.<br/><ul class=o_timeline_tracking_value_list><li>Reason<span> : </span><span class=o_timeline_tracking_value>%s</span></li></ul>") % (self.name, reason))
+            self.message_post(body=body, partner_ids=[self.employee_id.user_id.partner_id.id])
```

Comment:
I don't remember why, I think that it was initially in the specs, that only the employee should be warned. But in fact what you're saying makes sense.